src/js/components/TextInput/TextInput.js

```diff
@@ -258,104 +275,94 @@ const TextInput = forwardRef(
-                              adjustedEvent.target = inputRef.current;
-                              handleSuggestionSelect(adjustedEvent);
-                            }
-                            setValue(suggestion);
-                          }}
-                          onMouseOver={() => setActiveSuggestionIndex(index)}
-                          onFocus={() => setActiveSuggestionIndex(index)}
-                        >
-                          {child}
-                        </Button>
-                      </li>
+            <StyledSuggestions>
+              <InfiniteScroll items={suggestions} step={theme.select.step}>
+                {(suggestion, index, itemRef) => {
+                  // Determine whether the label is done as a child or
+                  // as an option Button kind property.
+                  const renderedLabel = renderLabel(suggestion);
+                  let child;
+                  if (typeof renderedLabel !== 'string')
+                    // must be an element rendered by suggestions.label
+                    child = renderedLabel;
+                  else if (!theme.button.option)
+                    // don't have theme support, need to layout here
+                    child = (
+                      <Box align="start" pad="small">
+                        {renderedLabel}
+                      </Box>
                     );
-                  }}
-                </InfiniteScroll>
-              </StyledSuggestions>
-            </ContainerBox>
-          </Drop>
-        </Keyboard>
+                  // if we have a child, turn on plain, and hoverIndicator
+
+                  return (
+                    <li
+                      key={`${stringLabel(suggestion)}-${index}`}
+                      ref={itemRef}
+                    >
+                      <Button
+                        active={activeSuggestionIndex === index}
+                        ref={r => {
+                          suggestionRefs[index] = r;
+                        }}
+                        fill
+                        plain={!child ? undefined : true}
+                        align="start"
+                        kind={!child ? 'option' : undefined}
+                        hoverIndicator={!child ? undefined : 'background'}
+                        label={!child ? renderedLabel : undefined}
+                        onClick={event => setFromSuggestion(event, suggestion)}
+                        onMouseOver={() => setActiveSuggestionIndex(index)}
+                        onFocus={() => setActiveSuggestionIndex(index)}
+                      >
+                        {child}
+                      </Button>
+                    </li>
+                  );
+                }}
+              </InfiniteScroll>
+            </StyledSuggestions>
+          </ContainerBox>
+        </Drop>
       );
     }
 
+    // If we have focus, then we set the target to the document, otherwise
```

Comment:
Why do we set the target to the document when we're in focus?

Reply:
`focus` being true doesn't mean the input has focus, it means that either the input or the drop has focus. We need to target the document to ensure we cover both contexts when the key is pressed.

Reply:
I've put more into the comment to attempt to clarify.